core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java

```diff
@@ -108,12 +129,26 @@ public Stream<InputSplit<List<CloudObjectLocation>>> createSplits(
   )
   {
     if (!CollectionUtils.isNullOrEmpty(objects)) {
-      return objects.stream().map(object -> new InputSplit<>(Collections.singletonList(object)));
+      if (StringUtils.isNotBlank(filter)) {
+        return objects.stream()
```

Comment:
Can you please shed some light into why do we require this change for **objects** and **uris**?
These inputs are controlled by the user. Shouldn't the user not provide the files if they do not need to ingested. 

The filtering makes sense if we provide a **prefix** as then we do not know what can be present in the folder.

For that, I think we should explore: https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html#API_ListObjectsV2_ResponseSyntax and similar for other connectors. 
We can read up on prefix-based s3 lists for some part of the filtering at least. 
LMK if I am missing a use case that I do not understand.